src/Security/Security.php

```diff
@@ -1138,6 +1202,23 @@ public static function database_is_ready()
     }
 
     /**
+     * Resets the database_is_ready cache
+     */
+    public static function clear_database_is_ready()
+    {
+        self::$database_is_ready = null;
+        self::$force_database_is_ready = null;
```

Comment:
clearing both these properties feels unexpected to me... When would this be used?